recipe/gen_patch_json.py

```diff
@@ -279,6 +279,10 @@ def _gen_new_index(repodata, subdir):
         if "ntl" in deps and record_name != "sage":
             _rename_dependency(fn, record, "ntl", "ntl 10.3.0")
 
+        deps = record.get("depends", ())
+        if "libspatialindex" in deps:
+            _rename_dependency(fn, record, "libspatialindex", "libspatialindex 1.8")
```

Comment:
I don't think this works. (The ntl code above is wrong too)

Reply:
:smile: 

No idea what is supposed to work here though.

Reply:
After a bit of testing, it seems like this will only apply a pin to libspatialindex 1.8 if the package is unpinned. So, in the future, if you want to be able to use libspatialindex 1.9 I think this will still be ok, provided that the dependency is pinned to be something like `> 1.8`.
I'm not entirely sure about pinning directly to 1.8 though, is there any case where a package will want libspatialindex < 1.8?

Reply:
> I'm not entirely sure about pinning directly to 1.8 though, is there any case where a package will want libspatialindex < 1.8?

Nope. `libspatialindex` was not updated in ages so we have mostly `1.8.5` and `1.9.0` only available.

Thanks for reviewing this @soapy1! Is it good to be merged? I'm not sure what are @isuruf's concerns here.

Reply:
If we had `libspatialindex 1.9` wouldn't this change it to `libspatialindex 1.8 1.8`

Reply:
Unless those `deps` are only unpinned then I guess it would be `libspatialindex 1.8 1.9` :-|